src/renderers/shared/server/ReactPartialRenderer.js

```diff
@@ -443,18 +445,36 @@ function resolve(child, context) {
   return {child, context};
 }
 
+type Frame = {
+  domNamespace: string,
+  children: any,
+  childIndex: number,
+  context: Object,
+  footer: string,
+  debugElementStack: Array<any>,
+};
+
 class ReactDOMServerRenderer {
-  constructor(element, makeStaticMarkup) {
+  stack: Array<Frame>;
+  exhausted: boolean;
+  currentSelectValue: null;
+  previousWasTextNode: boolean;
+  makeStaticMarkup: boolean;
+
+  constructor(
+    element: React.Element | Array<React.Element>,
+    makeStaticMarkup: boolean,
+  ) {
     var children = React.isValidElement(element) ? [element] : toArray(element);
-    var topFrame = {
+    var topFrame: Frame = ({
       // Assume all trees start in the HTML namespace (not totally true, but
       // this is what we did historically)
       domNamespace: Namespaces.html,
       children,
       childIndex: 0,
       context: emptyObject,
       footer: '',
-    };
+    }: any);
```

Comment:
Why?

Reply:
all of these are because of https://github.com/facebook/react/pull/11251#discussion_r145140768

```jsx
var frame = {
  ...everythingExcept_debugElementStack
};
if (process.env.NODE_ENV) {
  frame.debugElementStack = [];
}
```

So the `frame: Frame = ({}: any)` annotation is so that flow knows that this object can have the `debugElementStack` property.

```jsx
var frame: Frame = ({
  ...everythingExcept_debugElementStack
// the any is to prevent Flow from considered this a
// sealed object without the debugElementStack
}: any);
```

without this then the following chunks of code have a lot of errors due to `debugElementStack` being `undefined`

https://github.com/iamdustan/react/blob/bacc1e72f58e1ec3eab16c52d6db28c7d0c21014/src/renderers/shared/server/ReactPartialRenderer.js#L65-L98

Reply:
Typing something as `any` is generally as bad as not typing in the first place. Or possibly worse because now you have an illusion of type safety.

Could we perhaps use approach like

https://github.com/facebook/react/blob/90370f28ff1ce876c8ee6ec0062c921cf05b7924/src/renderers/dom/fiber/ReactDOMFiberEntry.js#L200-L201

and 

https://github.com/facebook/react/blob/90370f28ff1ce876c8ee6ec0062c921cf05b7924/src/renderers/dom/fiber/ReactDOMFiberEntry.js#L210

?

Or just treat that field as always optional?

Reply:
It is typed similarly, just `var binding: Type = ({}: any);` instead of `var binding = (({}: any): Type);`

Though I’ll look to into make the making separate `*Prod` and `*Dev` variants yet.